src/utilities/middleware.browser.spec.ts

```diff
@@ -2,50 +2,73 @@ import { expect, test, vi } from 'vitest'
 import { RouteMiddleware } from '@/types'
 import { RouterPushError, RouterRejectionError, RouterReplaceError } from '@/types/errors'
 import { Route } from '@/types/routes'
+import { RouterRoute } from '@/utilities/createRouterRoute'
+import { createRouterRouteQuery } from '@/utilities/createRouterRouteQuery'
 import { executeMiddleware } from '@/utilities/middleware'
-import { resolveRoutes } from '@/utilities/resolveRoutes'
 import { component } from '@/utilities/testHelpers'
 
 test('calls middleware with correct routes', () => {
   const middleware = vi.fn()
+
   const routeA = {
     name: 'routeA',
     path: '/routeA',
     component,
     middleware,
   } as const satisfies Route
 
+  const routerRouteA: RouterRoute = {
+    matched: routeA,
+    matches: [routeA],
+    name: routeA.name,
+    query: createRouterRouteQuery(),
+    params: {},
+  }
+
   const routeB = {
     name: 'routeB',
     path: '/routeB',
     component,
   } as const satisfies Route
 
-  const [resolvedA, resolvedB] = resolveRoutes([routeA, routeB])
+  const routerRouteB: RouterRoute = {
+    matched: routeB,
+    matches: [routeB],
+    name: routeB.name,
+    query: createRouterRouteQuery(),
+    params: {},
+  }
 
-  executeMiddleware({ to: resolvedA, from: resolvedB })
+  executeMiddleware({ to: routerRouteA, from: routerRouteB })
 
   expect(middleware).toHaveBeenCalledOnce()
 
   const [to, { from }] = middleware.mock.lastCall
-  expect(to).toMatchObject(resolvedA)
-  expect(from).toMatchObject(resolvedB)
+  expect(to).toMatchObject(routerRouteA!)
+  expect(from).toMatchObject(routerRouteB!)
 })
 
 test.each<{ type: string, error: any, middleware: RouteMiddleware }>([
   { type: 'reject', error: RouterRejectionError, middleware: (_to, { reject }) => reject('NotFound') },
   { type: 'push', error: RouterPushError, middleware: (_to, { push }) => push('') },
   { type: 'replace', error: RouterReplaceError, middleware: (_to, { replace }) => replace('') },
 ])('throws exception when $type is called', async ({ error, middleware }) => {
-  const routeA = {
+  const route = {
     name: 'routeA',
     path: '/routeA',
     component,
     middleware,
   } as const satisfies Route
 
-  const [resolvedA] = resolveRoutes([routeA])
-  const execute = (): Promise<void> => executeMiddleware({ to: resolvedA, from: null })
+  const routerRoute: RouterRoute = {
+    matched: route,
+    matches: [route],
+    name: route.name,
+    query: createRouterRouteQuery(),
+    params: {},
+  }
```

Comment:
`nit` do you think we should have an internal utility like `resolveRoutes`, but for turning a `Route` into `RouterRoute`? I've seen blocks like this in several places in several files

Reply:
something else to consider which tripped me up is that we have a thing called `createRouterRoute` but it doesn't do what I suggested, it creates `router.route` not `RouterRoute` 🤔

Reply:
We have many `createRouterX` methods that are factories for the various router properties. So this is consistent with that pattern. 

To create a `RouterRoute` you really need a resolved route AND the query and params. And in the codebase we only do that in one place (`getRouterRouteForUrl`). But in tests I created some manually so that way I could test functionality separately without depending on other utilities or services. I considered a test utility for this but ultimately decided the tests were simple enough.